### PR TITLE
docs: add docs for CST commands

### PR DESCRIPTION
**Pull request by spena**
### Description 
Added docs for source tables and streams.

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs/developer-guide/ksqldb-reference/create-stream.md [hunk 1]
```diff
@@ -20,6 +20,15 @@ Create a new stream with the specified columns and properties.
 If the IF NOT EXISTS clause is present, the statement won't fail if a 
 stream with the same name already exists.
 
+### Source streams
+
+A stream can be created as read-only if the SOURCE clause is provided. INSERTS statements and the
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/create-stream.md:25]:
```suggestion
A stream can be created as read-only if the SOURCE clause is provided. For a read-only stream, INSERT statements, and the

```

File: docs/developer-guide/ksqldb-reference/create-stream.md [hunk 2]
```diff
@@ -20,6 +20,15 @@ Create a new stream with the specified columns and properties.
 If the IF NOT EXISTS clause is present, the statement won't fail if a 
 stream with the same name already exists.
 
+### Source streams
+
+A stream can be created as read-only if the SOURCE clause is provided. INSERTS statements and the
+DELETE TOPIC on DROP STREAM statements won't be permitted. Source streams do not support pull
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/create-stream.md:26]:
```suggestion
DELETE TOPIC on DROP STREAM statements aren't permitted. Source streams don't support pull

```

File: docs/developer-guide/ksqldb-reference/create-stream.md [hunk 3]
```diff
@@ -20,6 +20,15 @@ Create a new stream with the specified columns and properties.
 If the IF NOT EXISTS clause is present, the statement won't fail if a 
 stream with the same name already exists.
 
+### Source streams
+
+A stream can be created as read-only if the SOURCE clause is provided. INSERTS statements and the
+DELETE TOPIC on DROP STREAM statements won't be permitted. Source streams do not support pull
+queries currently. Only source tables support running pull queries on them.
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/create-stream.md:27]:
```suggestion
queries. Only source tables support running pull queries on them.

```

File: docs/developer-guide/ksqldb-reference/create-stream.md [hunk 4]
```diff
@@ -20,6 +20,15 @@ Create a new stream with the specified columns and properties.
 If the IF NOT EXISTS clause is present, the statement won't fail if a 
 stream with the same name already exists.
 
+### Source streams
+
+A stream can be created as read-only if the SOURCE clause is provided. INSERTS statements and the
+DELETE TOPIC on DROP STREAM statements won't be permitted. Source streams do not support pull
+queries currently. Only source tables support running pull queries on them.
+
+To disable the SOURCE stream feature, you can set `ksql.source.table.materialization.enabled` to
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/create-stream.md:29]:
```suggestion
To disable the SOURCE stream feature, set `ksql.source.table.materialization.enabled` to

```

File: docs/developer-guide/ksqldb-reference/create-table.md [hunk 5]
```diff
@@ -20,6 +20,19 @@ Create a new table with the specified columns and properties.
 If the IF NOT EXISTS clause is present, the statement won't fail if a
 table with the same name already exists.
 
+### Source tables
+
+If the SOURCE clause is present, then pull queries can be executed on it. Note that the SOURCE
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/create-table.md:25]:
```suggestion
If the SOURCE clause is present, you can execute pull queries on it. The SOURCE

```

File: docs/developer-guide/ksqldb-reference/create-table.md [hunk 6]
```diff
@@ -20,6 +20,19 @@ Create a new table with the specified columns and properties.
 If the IF NOT EXISTS clause is present, the statement won't fail if a
 table with the same name already exists.
 
+### Source tables
+
+If the SOURCE clause is present, then pull queries can be executed on it. Note that the SOURCE
+clause will execute an internal query for this table in order to create a materialized state which
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/create-table.md:26]:
```suggestion
clause executes an internal query for the table to create a materialized state that's

```

File: docs/developer-guide/ksqldb-reference/create-table.md [hunk 7]
```diff
@@ -20,6 +20,19 @@ Create a new table with the specified columns and properties.
 If the IF NOT EXISTS clause is present, the statement won't fail if a
 table with the same name already exists.
 
+### Source tables
+
+If the SOURCE clause is present, then pull queries can be executed on it. Note that the SOURCE
+clause will execute an internal query for this table in order to create a materialized state which
+is used by pull queries. This query cannot be terminated manually. This query is only terminated
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/create-table.md:27]:
```suggestion
used by pull queries. You can't terminate this query manually. You can terminate it only

```

File: docs/developer-guide/ksqldb-reference/create-table.md [hunk 8]
```diff
@@ -20,6 +20,19 @@ Create a new table with the specified columns and properties.
 If the IF NOT EXISTS clause is present, the statement won't fail if a
 table with the same name already exists.
 
+### Source tables
+
+If the SOURCE clause is present, then pull queries can be executed on it. Note that the SOURCE
+clause will execute an internal query for this table in order to create a materialized state which
+is used by pull queries. This query cannot be terminated manually. This query is only terminated
+when dropping the table with DROP TABLE.
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/create-table.md:28]:
```suggestion
by dropping the table with the DROP TABLE statement.

```

File: docs/developer-guide/ksqldb-reference/create-table.md [hunk 9]
```diff
@@ -20,6 +20,19 @@ Create a new table with the specified columns and properties.
...
 table with the same name already exists.
 
+### Source tables
+
+If the SOURCE clause is present, then pull queries can be executed on it. Note that the SOURCE
+clause will execute an internal query for this table in order to create a materialized state which
+is used by pull queries. This query cannot be terminated manually. This query is only terminated
+when dropping the table with DROP TABLE.
+
+Also, the table will be created as read-only when the SOURCE clause is provided. INSERTS statements
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/create-table.md:30]:
```suggestion
Also, when you provide the SOURCE clause, the table is created as read-only. For a read-only table, INSERT statements

```

File: docs/developer-guide/ksqldb-reference/create-table.md [hunk 10]
```diff
@@ -20,6 +20,19 @@ Create a new table with the specified columns and properties.
...
+and the DELETE TOPIC on DROP TABLE statements won't be permitted.
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/create-table.md:31]:
```suggestion
and the DELETE TOPIC on DROP TABLE statements aren't permitted.

```

File: docs/developer-guide/ksqldb-reference/create-table.md [hunk 11]
```diff
@@ -20,6 +20,19 @@ Create a new table with the specified columns and properties.
...
+
+To disable the SOURCE table feature, you can set `ksql.source.table.materialization.enabled` to
+`false` in your KSQL server properties file.
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/create-table.md:34]:
```suggestion
`false` in your ksqlDB server properties file.

```

File: docs/developer-guide/ksqldb-reference/create-table.md [hunk 12]
```diff
@@ -20,6 +20,19 @@ Create a new table with the specified columns and properties.
...
+
+If the SOURCE clause is present, then pull queries can be executed on it. Note that the SOURCE
+clause will execute an internal query for this table in order to create a materialized state which
+is used by pull queries. This query cannot be terminated manually. This query is only terminated
+when dropping the table with DROP TABLE.
+
+Also, the table will be created as read-only when the SOURCE clause is provided. INSERTS statements
+and the DELETE TOPIC on DROP TABLE statements won't be permitted.
+
+To disable the SOURCE table feature, you can set `ksql.source.table.materialization.enabled` to
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/create-table.md:33]:
```suggestion
To disable the SOURCE table feature, set `ksql.source.table.materialization.enabled` to

```

**Review by JimGalasyn — APPROVED** [review]
LGTM, with a few copy edits.

File: docs/developer-guide/ksqldb-reference/create-stream.md (hunk 3 above)
Comment (jzaralim) [inline docs/developer-guide/ksqldb-reference/create-stream.md:27]: Pull queries don't work on any stream, so I'm not sure if we need to specify this here.

**Review by jzaralim — APPROVED** [review]

**Comment by spena** [comment]
Thanks @JimGalasyn . I applied all your suggestions manually and rebased the branch. This features is new in 0.22, so we don't need to cherry pick it to 7.0.x

**Pull request merged**